test/qiree/Module.test.cc

```diff
@@ -109,6 +109,27 @@ TEST_F(ModuleTest, several_gates)
     EXPECT_FALSE(flags.dynamic_result_management);
 }
 
+//---------------------------------------------------------------------------//
+TEST_F(ModuleTest, parse_ir_from_file) {
+
+    // Helper function to read a file and return its contents as a string to be fed into from_bytes
+    auto read_ll_file = [](const std::string& path) -> std::string {
+        std::ifstream file(path);
+        if (!file) throw std::runtime_error("Cannot open file: " + path);
+        std::ostringstream buf;
+        buf << file.rdbuf();
+        return buf.str();
+    };
+
+    // Read the LLVM IR from a file and parse it
+    std::string ir = read_ll_file(this->test_data_path("bell.ll"));
+
+    // Expect no exceptions during parsing
+    EXPECT_NO_THROW({
+        std::unique_ptr<Module> m = Module::from_bytes(ir);
+    });
```

Comment:
```suggestion
    std::unique_ptr<Module> m;
    // Expect no exceptions during parsing
    EXPECT_NO_THROW(m = Module::from_bytes(ir));
    ASSERT_TRUE(m);
```
and then maybe check that the flags loaded correctly. As is, if your static module function returned `nullptr`, it would pass the test :)